server/src/main/java/org/elasticsearch/action/admin/indices/template/get/GetComponentTemplateAction.java

```diff
@@ -198,12 +198,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 builder.field(NAME.getPreferredName(), componentTemplate.getKey());
                 builder.field(COMPONENT_TEMPLATE.getPreferredName());
                 componentTemplate.getValue()
-                    .toXContent(
-                        builder,
-                        DataStreamLifecycle.maybeAddEffectiveRetentionParams(params),
-                        rolloverConfiguration,
-                        globalRetention
-                    );
+                    .toXContent(builder, DataStreamLifecycle.maybeAddEffectiveRetentionParams(params), rolloverConfiguration);
```

Comment:
Does it still make sense to show the `effective_retention` at all for templates? The configured retention is already visible (if configured) in the `data_retention` field and since the `effective_retention` field will always only show the template's configured retention, it might actually be confusing if users see an `effective_retention` but it doesn't take the global retention into account. What do you guys think?


I don't immediately see the value of still showing the `effective_retention` field

Reply:
Yeah I think you're right. I'll remove that.

Reply:
Yeah, I think it's reasonable to remove it. I don't think templates should display anything for effective retention. They should only be displaying what's actually configured in the template itself. The data stream API is the place where effective retention should be displayed.